### ISSUE TITLE
Record "only errors" to unique log files

### ISSUE THREAD

**Issue by aliparlakci** [aliparlakci/bulk-downloader-for-reddit#252]
Logs should be preserved. But only the ones that are related to an internal issue of BDfR and can help the developer to debug it. As the logger.error messages are rare, they would not take much disk space. Other log levels in log_output.txt can be overridden for each run as they do not provide any sensible information.

**Comment by Serene-Arc** [comment]
I am strongly against this for reasons I have explained in the developer's chat, but, to sum up:

- Rotating log files would solve the space issue, which I am about to implement
- These logs might give occasionally vital information for obscure or complex bugs and the conditions that give rise to them
- Log files are easy to parse and filter with the formatting options that I have included as part of the program; since the level of each message is contained in the message, it is easy to filter certain levels out and isolate others
- Logging and keeping only errors would make debugging far, far more difficult since by their nature, they give a very incomplete view of the program. If you parse the logs with only levels of ERROR and above, it isn't as easy to debug and not all errors are, technically, actionable

I don't think there are any downsides to actually keeping full logs for, say, the last 5 runs of the program. Most users will never need or look at them, and those that do (like developers) will have an easy time filtering and parsing.
